py/emitinlinextensa.c

```diff
@@ -237,21 +216,58 @@ static const opcode_table_3arg_t opcode_table_3arg[] = {
     {MP_QSTR_bge, RRI8_B, ASM_XTENSA_CC_GE, 0},
     {MP_QSTR_bgeu, RRI8_B, ASM_XTENSA_CC_GEU, 0},
     {MP_QSTR_blt, RRI8_B, ASM_XTENSA_CC_LT, 0},
+    {MP_QSTR_bltu, RRI8_B, ASM_XTENSA_CC_LTU, 0},
     {MP_QSTR_bnall, RRI8_B, ASM_XTENSA_CC_NALL, 0},
     {MP_QSTR_bne, RRI8_B, ASM_XTENSA_CC_NE, 0},
     {MP_QSTR_bnone, RRI8_B, ASM_XTENSA_CC_NONE, 0},
 };
 
+// The index of the qstrs matches the CCZ condition value to be embedded into the opcode.
+static const qstr_short_t BCCZ_OPCODES[] = { MP_QSTR_beqz, MP_QSTR_bnez, MP_QSTR_bltz, MP_QSTR_bgez };
+
+#if MICROPY_EMIT_INLINE_XTENSA_UNCOMMON_OPCODES
+typedef struct _single_opcode_t {
+    qstr_short_t name;
+    uint16_t value;
+} single_opcode_t;
+
+static const single_opcode_t NOARGS_OPCODES[] = {
+    {MP_QSTR_dsync, 0x2030},
+    {MP_QSTR_esync, 0x2020},
+    {MP_QSTR_extw,  0x20D0},
+    {MP_QSTR_ill,   0x0000},
+    {MP_QSTR_isync, 0x2000},
+    {MP_QSTR_memw,  0x20C0},
+    {MP_QSTR_rsync, 0x2010},
+};
+#endif
+
 static void emit_inline_xtensa_op(emit_inline_asm_t *emit, qstr op, mp_uint_t n_args, mp_parse_node_t *pn_args) {
     size_t op_len;
     const char *op_str = (const char *)qstr_data(op, &op_len);
 
     if (n_args == 0) {
-        if (op == MP_QSTR_ret_n) {
+        if (op == MP_QSTR_ret_n || op == MP_QSTR_ret) {
```

Comment:
Is `ret` really the same as `ret_n`?

Reply:
According to Cadence's documentation it seems to be (page 565):

> **Description**
>
> `RET.N` is the same as `RET` in a 16-bit encoding. `RET` returns from a routine called by `CALL0` or `CALLX0`.